wallet/src/de/schildbach/wallet/ui/dashpay/ContactsFragment.kt

```diff
@@ -123,12 +136,18 @@ class ContactsFragment : Fragment(R.layout.fragment_contacts_root), TextWatcher,
                 if (initialSearch && (mode != MODE_VIEW_REQUESTS)) {
                     if (it.data == null || it.data.isEmpty() || it.data.find { u -> u.requestReceived } == null) {
                         empty_state_pane.visibility = View.VISIBLE
-                        contacts_pane.visibility = View.GONE
+                        search.visibility = View.GONE
+                        icon.visibility = View.GONE
                     } else {
+                        search.visibility = View.VISIBLE
+                        icon.visibility = View.VISIBLE
                         empty_state_pane.visibility = View.GONE
-                        contacts_pane.visibility = View.VISIBLE
                     }
                     initialSearch = false
+                } else {
+                    if (it.data == null || it.data.isEmpty()) {
+                        dashPayViewModel.searchUsernames(query, 3)
```

Comment:
if query is "" (empty) then we shouldn't execute the search.

```
I/DapiClient: [DefaultDispatcher-worker-3] getDocuments(H9AxLAvgxEpq72pDg41nsqR3bY5Cv9hTT6yZdKzY3PaE, domain, {limit=3, orderBy=[[normalizedLabel, asc]], where=[[normalizedParentDomainName, ==, dash], [normalizedLabel, startsWith, ]], startAt=0})
...
W/DapiClient: [DefaultDispatcher-worker-3] RPC failed with 34.216.205.76: Status{code=INVALID_ARGUMENT, description=Invalid query: should be equal to one of the allowed values, cause=null}: Metadata(server=nginx/1.19.3,date=Fri, 04 Dec 2020 20:41:46 GMT,content-type=application/grpc,content-length=0,errors=[{"name":"JsonSchemaValidationError","keyword":"enum","dataPath":".where[1][1]","schemaPath":"#/properties/where/items/oneOf/0/items/1/enum","params":{"allowedValues":["<","<=","==",">",">="]}},{"name":"JsonSchemaValidationError","keyword":"enum","dataPath":".where[1][0]","schemaPath":"#/properties/where/items/oneOf/1/items/0/enum","params":{"allowedValues":["$createdAt","$updatedAt"]}},{"name":"JsonSchemaValidationError","keyword":"const","dataPath":".where[1][1]","schemaPath":"#/properties/where/items/oneOf/2/items/1/const","params":{"allowedValue":"in"}},{"name":"JsonSchemaValidationError","keyword":"minLength","dataPath":".where[1][2]","schemaPath":"#/properties/where/items/oneOf/3/items/2/minLength","params":{"limit":1}},{"name":"JsonSchemaValidationError","keyword":"const","dataPath":".where[1][1]","schemaPath":"#/properties/where/items/oneOf/4/items/1/const","params":{"allowedValue":"elementMatch"}},{"name":"JsonSchemaValidationError","keyword":"const","dataPath":".where[1][1]","schemaPath":"#/properties/where/items/oneOf/5/items/1/const","params":{"allowedValue":"length"}},{"name":"JsonSchemaValidationError","keyword":"const","dataPath":".where[1][1]","schemaPath":"#/properties/where/items/oneOf/6/items/1/const","params":{"allowedValue":"contains"}},{"name":"JsonSchemaValidationError","keyword":"oneOf","dataPath":".where[1]","schemaPath":"#/properties/where/items/oneOf","params":{"passingSchemas":null}}])
..
E/Documents: [DefaultDispatcher-worker-3] Document creation: unable to get documents of H9AxLAvgxEpq72pDg41nsqR3bY5Cv9hTT6yZdKzY3PaE: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Invalid query: should be equal to one of the allowed values
E/DashPayViewModel: [DefaultDispatcher-worker-3] search usernames: INVALID_ARGUMENT: Invalid query: should be equal to one of the allowed values

```

Reply:
Fixed in https://github.com/dashevo/dash-wallet/pull/566/commits/1c633ae301fccf67b410a4ded440bcc3c56485a2

Reply:
Excellent.